lib/api/BackbeatAPI.js

```diff
@@ -705,6 +875,22 @@ class BackbeatAPI {
                 error: errors.MalformedPOSTRequest.customizeDescription(errMsg),
             };
         }
+        // filter duplicate version ids if any
+        const trackVersionIDs = {};
+        if (process.env.CI === 'true') {
+            // ignore filtering for tests
+            return { reqBody };
+        }
+        reqBody = reqBody.filter(entry => {
+            if (!entry.VersionId) {
+                return true;
+            } else if (!trackVersionIDs[entry.VersionId]) {
```

Comment:
As mentioned in another comment, I think it could be worthwhile to also check the bucket and object names here.

Minor: Let's make this another `if` block (rather than `else..if`) because we return in the first one.